internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1166,10 +1166,15 @@ func (r *Reconciler) reconcileRestoreJob(ctx context.Context,
 		"--pg1-path=" + pgdata,
 		"--repo=" + regexRepoIndex.FindString(repoName)}...)
 
+	targetRegex, err := regexp.Compile("--target[ =]")
+	if err != nil {
+		return err
+	}
+
 	var deltaOptFound, foundTarget bool
 	for _, opt := range opts {
 		switch {
-		case strings.Contains(opt, "--target"):
+		case targetRegex.Match([]byte(opt)):
```

Comment:
:pick: I think a comment would be good here. Something like, "don't allow the `--target` flag, but allow flags that start with `--target` such as `--target-timeline`" just to help clarify what's going on.